captum/_utils/sample_gradient.py

```diff
@@ -0,0 +1,176 @@
+from collections import defaultdict
+from enum import Enum
+from typing import Tuple, Union, cast, Iterable
+
+import torch
+from captum._utils.common import _format_tensor_into_tuples
+from torch import Tensor
+from torch.nn import Module
+
+
+def linear_param_grads(
+    module: Module, activation: Tensor, gradient_out: Tensor, reset: bool = False
```

Comment:
could this also be tuples of tensors instead of tensors ?

Reply:
For now, all supported modules (nn.Conv and nn.Linear) only operate on single tensors, so for now this should only be a tensor (we extract a single tensor even if a tuple is obtained from the hook). This line also only affects nn.Linear. If we need to add support for new modules which operate on multiple inputs, we should definitely look into this further.